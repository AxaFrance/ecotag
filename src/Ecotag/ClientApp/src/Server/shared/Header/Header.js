import React from 'react';
import {Header, Name, User} from '@axa-fr/react-toolkit-all';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import withAuthentication from '../../withAuthentication';
import i18next from "i18next";
import './Header.scss';

const formatRoles = (roles) => {
    return roles.reduce(
        (previousValue, currentValue) => {
            if (!previousValue) {
                return currentValue;
            }
            return `${previousValue}, ${currentValue}`;
        },
        ''
    );
}

export const AppHeader = ({user: {title = 'Ecotag', link = null, name = i18next.t('shared.header.errors.not_connected'), roles = ['NOT_FOUND']}}) => (
    <Header>
        <Name title={title} subtitle={i18next.t('shared.header.title')} img={logo} alt={title}/>
        <User name={name} href={link} profile={formatRoles(roles)}/>
    </Header>
);

export default withAuthentication()(AppHeader);
