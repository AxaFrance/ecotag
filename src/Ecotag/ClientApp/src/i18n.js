import i18n from 'i18next';
import {initReactI18next} from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import toolkit_en from './Toolkit/translations/en/toolkit.json';
import toolkit_fr from './Toolkit/translations/fr/toolkit.json';
import constants_dataset_en from './Server/Dataset/translations/en/constants.json';
import constants_dataset_fr from './Server/Dataset/translations/fr/constants.json'
import translations_dataset_en from './Server/Dataset/translations/en/translation.json';
import translations_dataset_fr from './Server/Dataset/translations/fr/translation.json';
import constants_group_en from './Server/Group/translations/en/constants.json';
import constants_group_fr from './Server/Group/translations/fr/constants.json';
import translations_group_en from './Server/Group/translations/en/translation.json';
import translations_group_fr from './Server/Group/translations/fr/translation.json';
import constants_project_en from './Server/Project/translations/en/constants.json';
import constants_project_fr from './Server/Project/translations/fr/constants.json';
import translations_project_en from './Server/Project/translations/en/translation.json';
import translations_project_fr from './Server/Project/translations/fr/translation.json';
import translations_home_en from './Server/Home/translations/en/translation.json';
import translations_home_fr from './Server/Home/translations/fr/translation.json';
import translations_not_found_en from './Server/NotFound/translations/en/translation.json';
import translations_not_found_fr from './Server/NotFound/translations/fr/translation.json';
import translations_shared_en from './Server/shared/translations/en/translation.json';
import translations_shared_fr from './Server/shared/translations/fr/translation.json';

const resources = {
    en: {
        translation: {
            ...translations_dataset_en,
            ...translations_group_en,
            ...translations_project_en,
            ...translations_home_en,
            ...translations_not_found_en,
            ...translations_shared_en
        },
        constants: {
            ...constants_dataset_en,
            ...constants_group_en,
            ...constants_project_en
        },
        toolkit: toolkit_en
    },
    fr: {
        translation: {
            ...translations_dataset_fr,
            ...translations_group_fr,
            ...translations_project_fr,
            ...translations_home_fr,
            ...translations_not_found_fr,
            ...translations_shared_fr
        },
        constants: {
            ...constants_dataset_fr,
            ...constants_group_fr,
            ...constants_project_fr
        },
        toolkit: toolkit_fr
    }
};

i18n
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
        resources,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false
        }
    });

export default i18n;
