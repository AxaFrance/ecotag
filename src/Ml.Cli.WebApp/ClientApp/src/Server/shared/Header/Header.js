import React from 'react';
import { Header, Name, User } from '@axa-fr/react-toolkit-all';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import withAuthentication from '../../withAuthentication';
import './Header.scss';

export const HeaderApp = ({ user: { title = 'Ecotag', link = null, name = 'Non ConnectÃ©', role = 'ADMIN' } }) => (
  <Header>
    <Name title={title} subtitle="An eco-system that aim to be eco for the planet" img={logo} alt={title} />
    <User name={name} href={link} profile={role} />
  </Header>
);

export default withAuthentication()(HeaderApp);
