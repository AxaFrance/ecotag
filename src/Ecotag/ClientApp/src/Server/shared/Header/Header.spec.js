import React from 'react';
import '@testing-library/jest-dom';
import {render} from '@testing-library/react';
import {AppHeader} from './Header';
import '../../../i18n';
import i18next from 'i18next';

describe('Header', () => {
    const user = {
        roles: ["youhou", "youhou"],
        name: 'Guillaume Chervet'
    };

    it('Renders header page in english', async () => {
        const {asFragment} = render(<AppHeader user={user}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('Renders header page in french', () => {
        i18next.changeLanguage('fr');
        const {asFragment} = render(<AppHeader user={user}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
