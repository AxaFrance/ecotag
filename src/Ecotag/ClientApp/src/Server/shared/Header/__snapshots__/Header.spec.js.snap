// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders header page in english 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Ecotag"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            Ecotag
            <span
              class="af-header__subtitle"
            >
              An eco-system to be eco-friendly
            </span>
          </h2>
        </div>
        <div
          class="af-info-user"
        >
          <span
            class="af-info-user__name"
          >
            Guillaume Chervet
          </span>
          <span
            class="af-info-user__profile"
          >
            [youhou, youhou]
          </span>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header Renders header page in french 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Ecotag"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            Ecotag
            <span
              class="af-header__subtitle"
            >
              Un écosystème pour être écologique
            </span>
          </h2>
        </div>
        <div
          class="af-info-user"
        >
          <span
            class="af-info-user__name"
          >
            Guillaume Chervet
          </span>
          <span
            class="af-info-user__profile"
          >
            [youhou, youhou]
          </span>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;
